security/pkg/server/ca/server.go

```diff
@@ -48,35 +48,41 @@ type Server struct {
 	hostnames      []string
 	forCA          bool
 	port           int
+	monitoring     monitoringMetrics
 }
 
 // HandleCSR handles an incoming certificate signing request (CSR). It does
 // proper validation (e.g. authentication) and upon validated, signs the CSR
 // and returns the resulting certificate. If not approved, reason for refusal
 // to sign is returned as part of the response object.
 func (s *Server) HandleCSR(ctx context.Context, request *pb.CsrRequest) (*pb.CsrResponse, error) {
+	s.monitoring.CSR.Inc()
 	caller := s.authenticate(ctx)
 	if caller == nil {
 		log.Warn("request authentication failure")
+		s.monitoring.AuthenticationError.Inc()
```

Comment:
I think we're good to move forward for now. This is for potential future improvements: would it better to have one metric and using its different labels for each type of error? These metrics users can just look at these errors together or break down by error (label value) if they want. Otherwise they have to find all metrics' name and sum them up.

(Above is assuming promethus behaves simliarly to google's internal monitoring sys).

Reply:
Yes. That's my plan. I've added a TODO in ca file to have Sign() return an error code. I will implement it later.